### ISSUE TITLE
Availability of source code

### ISSUE THREAD

**Issue by YoshuaNava** [Pfaeff/pitch_tracker#1]
Hi,
First of all, thank you for sharing this very cool tool.

I was wondering if you have plans to share the source code. I think it would be super cool to go a step further, and make it an audio plugin that you can load in e.g. Ableton/Reaper/Logic/etc.

Best,
Yoshua Nava

**Comment by Pfaeff** [comment]
Hi,

thank you. I added the source code to the repository.

I don't think this would work as a plugin, though. Since it's all python and very slow. Most of my code is just for the visuals, the pitch tracking itself is done using the aubio library. For this to be useful as a plugin, it would probably need to be rewritten in C/C++. And since Reaper has "ReaTune", which might also be available as a VST for other DAWs, there hasn't been a lot of motivation on my part to implement this. Also it wasn't running on my new computer and I stopped using it myself.

Greetings,
Pfaeff

**Comment by YoshuaNava** [comment]
Hi @Pfaeff,

Thank you for the prompt reply.

Are you also a musician? I'm quite interested in using this tool to practice bendings and vibrato for guitar.
I already use Ableton Live, with its [time vs. amplitude plot], and metronome, to practice hitting notes on time:

![Ableton practice](https://github.com/Pfaeff/pitch_tracker/assets/7463211/d476318e-4070-4646-998b-81ee48569682)

This has been incredibly useful for improving on the instrument, it provides 'closed-loop' feedback 🚀 🤘 

> And since Reaper has "ReaTune", which might also be available as a VST for other DAWs, there hasn't been a lot of motivation on my part to implement this.

Does ReaTune implement a real time pitch plot?

> For this to be useful as a plugin, it would probably need to be rewritten in C/C++.

I'm a C++ developer, I could try to give a hand if you are interested migrating it.


Best,
Yoshua

**Comment by Pfaeff** [comment]
Yes, I am a musician as well and I developed this mainly for singing practice.

ReaTune has a similar graph in its "Correction" tab when turning on "Automatic Pitch Correction". I personally don't like it, that's why I wrote own tool.

At the moment I don't have the time to port this. Maybe something could be thrown together using my code, a little bit of help from ChatGPT and a good C++ library that does the pitch tracking 😅. I don't really know anything about VST development, though. If someone would port it, I'd love to use it, though.
